### ISSUE TITLE
Support `bun install <pkg> {--d|--optional}`

### ISSUE THREAD

**Issue by colinhacks** [oven-sh/bun#1819]
### What steps can reproduce the bug?

```
$ bun install zod -d
Failed to parse argument due to unexpected single dash
```

**Comment by Electroid** [comment]
This is blocked until we update [`zig-clap`](https://github.com/Hejsil/zig-clap), which seems to have added support for lowecase flags.

**Comment by jwhear** [comment]
This is currently blocked by https://github.com/Hejsil/zig-clap/issues/84, which in turn is blocked by https://github.com/ziglang/zig/issues/13779

**Comment by Electroid** [comment]
@jwhear Could we still workaround the issue if we provide a long name for every flag? Do we have any flags that are single char only?

**Comment by jwhear** [comment]
> @jwhear Could we still workaround the issue if we provide a long name for every flag? Do we have any flags that are single char only?

Probably, I'm working on that now.  The only parameter that was short-form only is
 ```
-i          Automatically install dependencies and use global cache in bun's runtime, equivalent to --install=fallback
```
I am adding long-form `--install-fallback` for now.

**Comment by Parzival-3141** [comment]
This seems to be fixed as of 4b63ced72.
```sh
> bun-debug install zod -d                                       
bun add v0.8.2_debug (4b63ced7)
  🔍 Resolving [1/1] [uws] connect(registry.npmjs.org, 443)
  🔒 Saving lockfile...

 installed zod@3.22.2


 1 packages installed [159.00ms]
```

```json
"devDependencies": {
    "zod": "^3.22.2"
  }
```

It also works for `-D`, `--dev`, and `--optional`, and when passing package URLs.
